src/lib_ccx/ccx_decoders_708.c

```diff
@@ -56,921 +48,1128 @@ const char *COMMANDS_C0[32]=
 {
-	{CW0,"CW0","SetCurrentWindow0",     1},
-	{CW1,"CW1","SetCurrentWindow1",     1},
-	{CW2,"CW2","SetCurrentWindow2",     1},
-	{CW3,"CW3","SetCurrentWindow3",     1},
-	{CW4,"CW4","SetCurrentWindow4",     1},
-	{CW5,"CW5","SetCurrentWindow5",     1},
-	{CW6,"CW6","SetCurrentWindow6",     1},
-	{CW7,"CW7","SetCurrentWindow7",     1},
-	{CLW,"CLW","ClearWindows",          2},
-	{DSW,"DSW","DisplayWindows",        2},
-	{HDW,"HDW","HideWindows",           2},
-	{TGW,"TGW","ToggleWindows",         2},
-	{DLW,"DLW","DeleteWindows",         2},
-	{DLY,"DLY","Delay",                 2},
-	{DLC,"DLC","DelayCancel",           1},
-	{RST,"RST","Reset",                 1},
-	{SPA,"SPA","SetPenAttributes",      3},
-	{SPC,"SPC","SetPenColor",           4},
-	{SPL,"SPL","SetPenLocation",        3},
-	{RSV93,"RSV93","Reserved",          1},
-	{RSV94,"RSV94","Reserved",          1},
-	{RSV95,"RSV95","Reserved",          1},
-	{RSV96,"RSV96","Reserved",          1},
-	{SWA,"SWA","SetWindowAttributes",   5},
-	{DF0,"DF0","DefineWindow0",         7},
-	{DF1,"DF0","DefineWindow1",         7},
-	{DF2,"DF0","DefineWindow2",         7},
-	{DF3,"DF0","DefineWindow3",         7},
-	{DF4,"DF0","DefineWindow4",         7},
-	{DF5,"DF0","DefineWindow5",         7},
-	{DF6,"DF0","DefineWindow6",         7},
-	{DF7,"DF0","DefineWindow7",         7}
+	{CW0, "CW0", "SetCurrentWindow0",     1},
+	{CW1, "CW1", "SetCurrentWindow1",     1},
+	{CW2, "CW2", "SetCurrentWindow2",     1},
+	{CW3, "CW3", "SetCurrentWindow3",     1},
+	{CW4, "CW4", "SetCurrentWindow4",     1},
+	{CW5, "CW5", "SetCurrentWindow5",     1},
+	{CW6, "CW6", "SetCurrentWindow6",     1},
+	{CW7, "CW7", "SetCurrentWindow7",     1},
+	{CLW, "CLW", "ClearWindows",          2},
+	{DSW, "DSW", "DisplayWindows",        2},
+	{HDW, "HDW", "HideWindows",           2},
+	{TGW, "TGW", "ToggleWindows",         2},
+	{DLW, "DLW", "DeleteWindows",         2},
+	{DLY, "DLY", "Delay",                 2},
+	{DLC, "DLC", "DelayCancel",           1},
+	{RST, "RST", "Reset",                 1},
+	{SPA, "SPA", "SetPenAttributes",      3},
+	{SPC, "SPC", "SetPenColor",           4},
+	{SPL, "SPL", "SetPenLocation",        3},
+	{RSV93, "RSV93", "Reserved",          1},
+	{RSV94, "RSV94", "Reserved",          1},
+	{RSV95, "RSV95", "Reserved",          1},
+	{RSV96, "RSV96", "Reserved",          1},
+	{SWA, "SWA", "SetWindowAttributes",   5},
+	{DF0, "DF0", "DefineWindow0",         7},
+	{DF1, "DF1", "DefineWindow1",         7},
+	{DF2, "DF2", "DefineWindow2",         7},
+	{DF3, "DF3", "DefineWindow3",         7},
+	{DF4, "DF4", "DefineWindow4",         7},
+	{DF5, "DF5", "DefineWindow5",         7},
+	{DF6, "DF6", "DefineWindow6",         7},
+	{DF7, "DF7", "DefineWindow7",         7}
 };
 
+//---------------------------------- HELPERS ------------------------------------
 
-cc708_service_decoder decoders[63]; // For primary and secondary languages
-
-void clear_packet(void)
+void _dtvcc_clear_packet(void)
```

Comment:
pass the structure dtvcc_ctx as argumnet, if possible try making function re-entrant